diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index f232ff9..7db83c2 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -6493,7 +6493,7 @@ qtdemux_parse_samples (GstQTDemux * qtdemux, QtDemuxStream * stream, guint32 n)
         cur->keyframe = TRUE;
         cur++;

+        stream->stco_sample_index += stream->samples_per_chunk * stream->sample_size;
-        stream->stco_sample_index += stream->samples_per_chunk;
       }
       stream->stsc_chunk_index = j;
     } else { 
