diff -rupN nose-1.3.7/distribute_setup.py nose/distribute_setup.py
--- nose-1.3.7/distribute_setup.py	2014-03-29 04:20:27.000000000 -0600
+++ nose/distribute_setup.py	2017-10-31 11:04:38.794504824 -0600
@@ -50,7 +50,7 @@ except ImportError:
         return os.spawnl(os.P_WAIT, sys.executable, *args) == 0
 
 DEFAULT_VERSION = "0.6.35"
-DEFAULT_URL = "http://pypi.python.org/packages/source/d/distribute/"
+DEFAULT_URL = "https://pypi.python.org/packages/source/d/distribute/"
 SETUPTOOLS_FAKED_VERSION = "0.6c11"
 
 SETUPTOOLS_PKG_INFO = """\
