diff -rupN libfslvpuwrap-1.0.58/vpu_wrapper.c libfslvpuwrap/vpu_wrapper.c
--- libfslvpuwrap-1.0.58/vpu_wrapper.c	2015-03-05 00:45:13.000000000 -0700
+++ libfslvpuwrap/vpu_wrapper.c	2019-08-05 10:39:13.727093116 -0600
@@ -203,9 +203,9 @@ static int g_seek_dump=DUMP_ALL_DATA;	/*
 #else
 #define VPU_BITS_BUF_SIZE		(1024*1024)
 #endif
-#define VPU_SLICE_SAVE_SIZE	0x17E800		//worst slice buffer size: 1920*1088*1.5/2= 1.5MB
+#define VPU_SLICE_SAVE_SIZE	0x2a3000		//worst slice buffer size: 1920*1920*1.5/2= 2.7MB
 #define VPU_PS_SAVE_SIZE		0x80000
-#define VPU_VP8_MBPARA_SIZE	0x87780			//68 * (1920 * 1088 / 256)=0x87780;
+#define VPU_VP8_MBPARA_SIZE	0x979200			//68 * (1920 * 1920 / 256)=0x87780;
 
 #define VPU_MIN_INIT_SIZE		(512)			//min required data size for seq init
 #define VPU_MIN_DEC_SIZE		(64*1024)		//min required data size for decode
@@ -5963,7 +5963,7 @@ VpuDecRetCode VPU_DecRegisterFrameBuffer
 	sBufInfo.avcSliceBufInfo.sliceSaveBufferSize=VPU_SLICE_SAVE_SIZE;
 #endif	
 //#if 1 //for iMX6: Vp8
-#if 0	//we set maxDecFrmInfo with 0 : in order to support dynamic resolution
+#if 1	//we set maxDecFrmInfo with 0 : in order to support dynamic resolution
 	sBufInfo.maxDecFrmInfo.maxMbX=(pVpuObj->obj.initInfo.nPicWidth+15)/16;
 	if(pVpuObj->obj.initInfo.nInterlace)
 	{
