diff -uNr openssl-1.1.1g.old/crypto/chacha/asm/chacha-armv8.pl openssl-1.1.1g/crypto/chacha/asm/chacha-armv8.pl
--- openssl-1.1.1g.old/crypto/chacha/asm/chacha-armv8.pl	2020-04-21 06:22:39.000000000 -0600
+++ openssl-1.1.1g/crypto/chacha/asm/chacha-armv8.pl	2021-04-14 09:59:08.860636258 -0600
@@ -125,6 +125,7 @@
 .text
 
 .extern	OPENSSL_armcap_P
+.hidden OPENSSL_armcap_P
 
 .align	5
 .Lsigma:
diff -uNr openssl-1.1.1g.old/crypto/poly1305/asm/poly1305-armv8.pl openssl-1.1.1g/crypto/poly1305/asm/poly1305-armv8.pl
--- openssl-1.1.1g.old/crypto/poly1305/asm/poly1305-armv8.pl	2020-04-21 06:22:39.000000000 -0600
+++ openssl-1.1.1g/crypto/poly1305/asm/poly1305-armv8.pl	2021-04-14 09:59:08.864636348 -0600
@@ -57,10 +57,14 @@
 
 // forward "declarations" are required for Apple
 .extern	OPENSSL_armcap_P
+.hidden OPENSSL_armcap_P
+.globl poly1305_init
+.hidden poly1305_init
 .globl	poly1305_blocks
+.hidden poly1305_blocks
 .globl	poly1305_emit
+.hidden poly1305_emit
 
-.globl	poly1305_init
 .type	poly1305_init,%function
 .align	5
 poly1305_init:
diff -uNr openssl-1.1.1g.old/crypto/sha/asm/sha1-armv8.pl openssl-1.1.1g/crypto/sha/asm/sha1-armv8.pl
--- openssl-1.1.1g.old/crypto/sha/asm/sha1-armv8.pl	2020-04-21 06:22:39.000000000 -0600
+++ openssl-1.1.1g/crypto/sha/asm/sha1-armv8.pl	2021-04-14 09:59:08.888636890 -0600
@@ -176,6 +176,8 @@
 .text
 
 .extern	OPENSSL_armcap_P
+.hidden OPENSSL_armcap_P
+
 .globl	sha1_block_data_order
 .type	sha1_block_data_order,%function
 .align	6
diff -uNr openssl-1.1.1g.old/crypto/sha/asm/sha512-armv8.pl openssl-1.1.1g/crypto/sha/asm/sha512-armv8.pl
--- openssl-1.1.1g.old/crypto/sha/asm/sha512-armv8.pl	2020-04-21 06:22:39.000000000 -0600
+++ openssl-1.1.1g/crypto/sha/asm/sha512-armv8.pl	2021-04-14 09:59:46.445484502 -0600
@@ -193,6 +193,8 @@
 .text
 
 .extern	OPENSSL_armcap_P
+.hidden        OPENSSL_armcap_P
+
 .globl	$func
 .type	$func,%function
 .align	6
@@ -840,11 +842,6 @@
 ___
 }
 
-$code.=<<___;
-#ifndef	__KERNEL__
-.comm	OPENSSL_armcap_P,4,4
-#endif
-___
 
 {   my  %opcode = (
 	"sha256h"	=> 0x5e004000,	"sha256h2"	=> 0x5e005000,
