From 351206e515fac7b71ad7bd5bcaa15c177c5f3e43 Mon Sep 17 00:00:00 2001
From: Paul Willoughby
Date: Mon, 29 Jan 2018 10:01:49 -0700
Subject: [PATCH] allow bootchartd to run longer than one minute
 
---
 init/bootchartd.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)
 
diff --git a/init/bootchartd.c b/init/bootchartd.c
index 5a1b3e8..747aa40 100644
--- a/init/bootchartd.c
+++ b/init/bootchartd.c
@@ -220,8 +220,10 @@ static void do_logging(unsigned sample_period_us)
    FILE *proc_diskstats = xfopen("proc_diskstats.log", "w");
    //FILE *proc_netdev = xfopen("proc_netdev.log", "w");
    FILE *proc_ps = xfopen("proc_ps.log", "w");
-   int look_for_login_process = (getppid() == 1);
-   unsigned count = 60*1000*1000 / sample_period_us; /* ~1 minute */
+   //don't stop when getty starts, keep running until count expires
+   //int look_for_login_process = (getppid() == 1);
+   int look_for_login_process = 0;
+   unsigned count = 6*60*1000*1000 / sample_period_us; /* ~6 minutes */
  
    while (--count && !bb_got_signal) {
        char *p;
-- 
2.14.1
