--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -31,15 +31,15 @@
 #include "rate.h"
 #include "led.h"
 
-#define IEEE80211_AUTH_TIMEOUT		(HZ / 5)
+#define IEEE80211_AUTH_TIMEOUT		(HZ / 2)
 #define IEEE80211_AUTH_TIMEOUT_LONG	(HZ / 2)
 #define IEEE80211_AUTH_TIMEOUT_SHORT	(HZ / 10)
-#define IEEE80211_AUTH_MAX_TRIES	3
-#define IEEE80211_AUTH_WAIT_ASSOC	(HZ * 5)
-#define IEEE80211_ASSOC_TIMEOUT		(HZ / 5)
+#define IEEE80211_AUTH_MAX_TRIES	16
+#define IEEE80211_AUTH_WAIT_ASSOC	(HZ * 10)
+#define IEEE80211_ASSOC_TIMEOUT		(HZ / 2)
 #define IEEE80211_ASSOC_TIMEOUT_LONG	(HZ / 2)
 #define IEEE80211_ASSOC_TIMEOUT_SHORT	(HZ / 10)
-#define IEEE80211_ASSOC_MAX_TRIES	3
+#define IEEE80211_ASSOC_MAX_TRIES	16
 
 static int max_nullfunc_tries = 2;
 module_param(max_nullfunc_tries, int, 0644);
