diff -rupN imx-gst1.0-plugin/plugins/aiurdemux/aiurdemux.c imx-gst1.0-plugin-4.0.9/plugins/aiurdemux/aiurdemux.c
--- imx-gst1.0-plugin/plugins/aiurdemux/aiurdemux.c	2019-02-26 15:51:17.352439675 -0700
+++ imx-gst1.0-plugin-4.0.9/plugins/aiurdemux/aiurdemux.c	2019-03-05 15:43:40.330409668 -0700
@@ -2244,6 +2244,15 @@ static void aiurdemux_parse_audio (GstAi
             stream->info.audio.block_align, stream->info.audio.sample_width,
             stream->bitrate);
         break;
+      case AUDIO_PCM_MULAW:
+        codec_mime = "audio/x-mulaw";
+        codec = "MULAW";
+        mime =
+            g_strdup_printf
+            ("%s, channels=(int)%ld, rate=(int)%ld, bitrate=(int)%ld", codec_mime,
+            stream->info.audio.n_channels, stream->info.audio.rate,
+            stream->bitrate);
+        break;
       case AUDIO_PCM:
       {
         int width, depth, endian;
Binary files imx-gst1.0-plugin/plugins/aiurdemux/.aiurdemux.c.swp and imx-gst1.0-plugin-4.0.9/plugins/aiurdemux/.aiurdemux.c.swp differ
