diff -rupN imx-gst1.0-plugin-4.0.9/ext-includes/fsl_media_types.h imx-gst1.0-plugin/ext-includes/fsl_media_types.h
--- imx-gst1.0-plugin-4.0.9/ext-includes/fsl_media_types.h	2016-03-10 08:08:42.000000000 -0700
+++ imx-gst1.0-plugin/ext-includes/fsl_media_types.h	2019-03-06 10:59:47.217498217 -0700
@@ -101,6 +101,7 @@ typedef enum
     AUIDO_SPEEX,
     AUDIO_REAL, /* Real audio types */
     AUDIO_EC3,
+    AUDIO_OPUS,
 }AudioCodecType;
 
 
diff -rupN imx-gst1.0-plugin-4.0.9/plugins/aiurdemux/aiurdemux.c imx-gst1.0-plugin/plugins/aiurdemux/aiurdemux.c
--- imx-gst1.0-plugin-4.0.9/plugins/aiurdemux/aiurdemux.c	2019-03-05 16:59:24.397555684 -0700
+++ imx-gst1.0-plugin/plugins/aiurdemux/aiurdemux.c	2019-03-06 11:02:50.083421233 -0700
@@ -2413,6 +2413,14 @@ static void aiurdemux_parse_audio (GstAi
             stream->info.audio.n_channels, stream->info.audio.rate,
             stream->bitrate);
         break;
+      case AUDIO_OPUS:
+        codec = "OPUS";
+        mime =
+            g_strdup_printf
+            ("audio/x-opus, channel-mapping-family=(int)0, channels=(int)%ld, rate=(int)%ld, bitrate=(int)%ld",
+            stream->info.audio.n_channels, stream->info.audio.rate,
+            stream->bitrate);
+        break;
        default:
         break;
     }
