diff -rupN python-dateutil-2.7.2/setup.py python-dateutil/setup.py
--- python-dateutil-2.7.2/setup.py	2018-03-26 07:32:32.000000000 -0600
+++ python-dateutil/setup.py	2021-04-13 11:51:03.932945464 -0600
@@ -10,6 +10,20 @@ from distutils.version import LooseVersi
 import warnings
 
 import io
+import re
+
+def read(*names, **kwargs):
+    with open(os.path.join( *names), 'r') as fp:
+        return fp.read()
+
+def find_version(*file_paths):
+    version_file = read(*file_paths)
+    version_match = re.search(r"^version = ['\']([^'\"]*)['\']",
+                              version_file, re.M)
+
+    if version_match:
+        return version_match.group(1)
+    raise RuntimeError("Unable to find version string.")
 
 if isfile("MANIFEST"):
     os.unlink("MANIFEST")
@@ -46,9 +60,6 @@ def README():
 README = README()
 
 setup(name="python-dateutil",
-      use_scm_version={
-          'write_to': 'dateutil/_version.py',
-      },
       description="Extensions to the standard Python datetime module",
       author="Gustavo Niemeyer",
       author_email="gustavo@niemeyer.net",
@@ -63,7 +74,7 @@ setup(name="python-dateutil",
       package_data={"dateutil.zoneinfo": ["dateutil-zoneinfo.tar.gz"]},
       zip_safe=True,
       requires=["six"],
-      setup_requires=['setuptools_scm'],
+      version=find_version("dateutil", "_version.py")
       install_requires=["six >=1.5"],  # XXX fix when packaging is sane again
       classifiers=[
           'Development Status :: 5 - Production/Stable',
