diff -ruN speedtest-cli.orig/speedtest_cli.py speedtest-cli/speedtest_cli.py
--- speedtest-cli.orig/speedtest_cli.py	2015-02-26 11:40:29.000000000 -0700
+++ speedtest-cli/speedtest_cli.py	2015-03-09 11:24:32.191012647 -0600
@@ -669,12 +669,12 @@
         try:
             unicode()
             print_(('Hosted by %(sponsor)s (%(name)s) [%(d)0.2f km]: '
-                   '%(latency)s ms' % best).encode('utf-8', 'ignore'))
+                   '%(latency)0.2f ms' % best).encode('utf-8', 'ignore'))
         except NameError:
             print_('Hosted by %(sponsor)s (%(name)s) [%(d)0.2f km]: '
-                   '%(latency)s ms' % best)
+                   '%(latency)0.2f ms' % best)
     else:
-        print_('Ping: %(latency)s ms' % best)
+        print_('Ping: %(latency)0.2f ms' % best)
 
     sizes = [350, 500, 750, 1000, 1500, 2000, 2500, 3000, 3500, 4000]
     urls = []
