diff -rupN git://systools/udprelay/CMakeLists.txt gita/systools/udprelay/CMakeLists.txt
--- git/systools/udprelay/CMakeLists.txt	2021-10-18 13:25:40.408507671 -0600
+++ gita/systools/udprelay/CMakeLists.txt	2021-10-18 13:18:13.055309923 -0600
@@ -1,6 +1,6 @@
 ExternalProject_Add(
   libpcap 
-  URL http://www.tcpdump.org/release/libpcap-1.5.3.tar.gz
+  URL https://www.tcpdump.org/release/libpcap-1.5.3.tar.gz
   PATCH_COMMAND sed -i "s/IFNAMSIZ/sizeof(int)/" pcap-linux.c
   CONFIGURE_COMMAND ./configure CC=${CMAKE_C_COMPILER} LDFLAGS=${CMAKE_C_FLAGS} CFLAGS=${CMAKE_C_FLAGS}  --host=${CONFIGURE_HOST} --with-pcap=linux --enable-canusb=no --enable-dbus=no --with-flex --with-bison --without-libnl --prefix=${CMAKE_BINARY_DIR}/stage
   BUILD_IN_SOURCE 1
