diff --git a/zipgateway-7.13.01-Source/usr/local/CMakeLists.txt b/zipgateway-7.13.01-Source/usr/local/CMakeLists.txt
index 165c6eb..2ef4cad 100644
--- a/zipgateway-7.13.01-Source/usr/local/CMakeLists.txt
+++ b/zipgateway-7.13.01-Source/usr/local/CMakeLists.txt
@@ -37,7 +37,7 @@ if(NOT CMAKE_CONFIGURATION_TYPES)
 endif()
 
 include( ExternalProject )
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -Werror -Wparentheses -Wno-array-bounds -Wno-unused-result -std=gnu99 -D_DEFAULT_SOURCE")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -Werror -Wparentheses -Wno-array-bounds -Wno-unused-result -std=gnu99 -D_DEFAULT_SOURCE -Wno-error=address-of-packed-member")
 set(CONFIGURE_HOST arm-linux)
 
 if(DEFINED ENABLE_CODE_COVERAGE)
diff --git a/zipgateway-7.13.01-Source/usr/local/contiki/platform/linux/parse_config.c b/zipgateway-7.13.01-Source/usr/local/contiki/platform/linux/parse_config.c
index 2c4cdfc..ddd003c 100644
--- a/zipgateway-7.13.01-Source/usr/local/contiki/platform/linux/parse_config.c
+++ b/zipgateway-7.13.01-Source/usr/local/contiki/platform/linux/parse_config.c
@@ -35,7 +35,7 @@ extern int optind, opterr, optopt;
 extern int prog_argc;
 extern char** prog_argv;
 
-const char* linux_conf_eeprom_file;
+extern const char* linux_conf_eeprom_file;
 const char* linux_conf_provisioning_cfg_file;
 const char* linux_conf_provisioning_list_storage_file;
 const char* linux_conf_tun_script;
diff --git a/zipgateway-7.13.01-Source/usr/local/src/analyzer/extract_get.py b/zipgateway-7.13.01-Source/usr/local/src/analyzer/extract_get.py
index c0d1468..ba3a5e6 100755
--- a/zipgateway-7.13.01-Source/usr/local/src/analyzer/extract_get.py
+++ b/zipgateway-7.13.01-Source/usr/local/src/analyzer/extract_get.py
@@ -31,7 +31,7 @@ if __name__ == '__main__':
 
   
   for c in x.getElementsByTagName("cmd_class"):
-  	cls = int(c.getAttribute("key"),16)
+        cls = int(c.getAttribute("key"),16)
         for q in c.childNodes:
              if(q.nodeType==1 and q.tagName=="cmd"):
                  cmd = int(q.getAttribute("key"),16)
diff --git a/zipgateway-7.13.01-Source/usr/local/src/zgw_backup.c b/zipgateway-7.13.01-Source/usr/local/src/zgw_backup.c
index ed431bb..ad4bf16 100644
--- a/zipgateway-7.13.01-Source/usr/local/src/zgw_backup.c
+++ b/zipgateway-7.13.01-Source/usr/local/src/zgw_backup.c
@@ -65,10 +65,10 @@ static void send_backup_started_to_script()
   send_command_to_backup_script("backup started");
 }
 
-char manifest_file[300];
+char manifest_file[PATH_MAX+10];
 static int copy_to_bkup_dir(const char* file)
 {
-  char cmd[PATH_MAX];
+  char cmd[PATH_MAX+20];
   snprintf(cmd,sizeof(cmd), "cp %s %s", file, bkup_dir); 
   system(cmd); 
   return 1;
@@ -109,7 +109,7 @@ static int record_to_manifest(const char* varname, const char* path)
 
 static int make_backup()
 {
-  char buff[512];
+  char buff[PATH_MAX+10];
   uint8_t chip_library;
   const char *sdk_ver;
   const char *nvm_id;
diff --git a/zipgateway-7.13.01-Source/usr/local/systools/libs/json_helpers/json_helpers.c b/zipgateway-7.13.01-Source/usr/local/systools/libs/json_helpers/json_helpers.c
index d682ab3..eec18ec 100644
--- a/zipgateway-7.13.01-Source/usr/local/systools/libs/json_helpers/json_helpers.c
+++ b/zipgateway-7.13.01-Source/usr/local/systools/libs/json_helpers/json_helpers.c
@@ -242,7 +242,7 @@ static bool search_object(json_object *jso,
 /*****************************************************************************/
 const char * json_get_object_path(const json_object* obj)
 {
-  static char path[300];
+  static char path[501];
   static json_path_item_t path_items[MAX_PATH_ITEM_COUNT];
 
   memset(path_items, 0, sizeof(path_items));
diff --git a/zipgateway-7.13.01-Source/usr/local/systools/udprelay/CMakeLists.txt b/zipgateway-7.13.01-Source/usr/local/systools/udprelay/CMakeLists.txt
index 62093a9..bb18c11 100644
--- a/zipgateway-7.13.01-Source/usr/local/systools/udprelay/CMakeLists.txt
+++ b/zipgateway-7.13.01-Source/usr/local/systools/udprelay/CMakeLists.txt
@@ -1,6 +1,7 @@
 ExternalProject_Add(
   libpcap 
   URL http://www.tcpdump.org/release/libpcap-1.5.3.tar.gz
+  PATCH_COMMAND sed -i "s/IFNAMSIZ/sizeof(int)/" pcap-linux.c
   CONFIGURE_COMMAND ./configure CC=${CMAKE_C_COMPILER} LDFLAGS=${CMAKE_C_FLAGS} CFLAGS=${CMAKE_C_FLAGS}  --host=${CONFIGURE_HOST} --with-pcap=linux --enable-canusb=no --enable-dbus=no --with-flex --with-bison --without-libnl --prefix=${CMAKE_BINARY_DIR}/stage
   BUILD_IN_SOURCE 1
 )
diff --git a/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgw_data.c b/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgw_data.c
index 9422d35..4003cd2 100644
--- a/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgw_data.c
+++ b/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgw_data.c
@@ -16,9 +16,6 @@
 #include "ZW_classcmd.h"
 #include "RD_internal.h" /* For  DEFAULT_WAKE_UP_INTERVAL (70 * 60) */
 
-uint32_t homeID = 0;
-uint8_t MyNodeID = 0;
-
 zip_gateway_backup_data_t backup_rep;
 
 /*
diff --git a/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgwr_eeprom.c b/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgwr_eeprom.c
index 0f30243..5ffd985 100644
--- a/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgwr_eeprom.c
+++ b/zipgateway-7.13.01-Source/usr/local/systools/zgw_restore/zgwr_eeprom.c
@@ -20,7 +20,7 @@ extern uint8_t MyNodeID;
 extern uint32_t homeID;
 
 /* Used by eeprom_init() implemented in contiki/platform/linux/File_eeprom.c */
-const char* linux_conf_eeprom_file;
+extern const char* linux_conf_eeprom_file;
 /* Path to restore the eeprom file */
 char* eeprom_fullpath;
 
