--- gst-plugins-good-1.4.1/gst/rtsp/gstrtspsrc.c	2014-07-21 02:24:27.000000000 -0600
+++ gst-plugins-good-1.4.1.a/gst/rtsp/gstrtspsrc.c	2017-04-05 00:28:40.000000000 -0600
@@ -6957,6 +6957,8 @@
 
   GST_DEBUG_OBJECT (src, "TEARDOWN...");
 
+  // stop connection flush so we can send TEARDOWN
+  gst_rtspsrc_connection_flush (src, FALSE);
   gst_rtspsrc_set_state (src, GST_STATE_READY);
 
   if (src->state < GST_RTSP_STATE_READY) {
@@ -7787,7 +7789,7 @@
   GST_OBJECT_UNLOCK (src);
 
   /* ensure synchronously all is closed and clean */
-  gst_rtspsrc_close (src, FALSE, TRUE);
+  gst_rtspsrc_close (src, FALSE, FALSE);
 
   return TRUE;
 }
