diff -ur qtgstreamer-orig/git-r0/git/src/QGst/ghostpad.cpp qtgstreamer/git-r0/git/src/QGst/ghostpad.cpp
--- a/src/QGst/ghostpad.cpp	2015-08-24 15:12:24.591417081 -0600
+++ b/src/QGst/ghostpad.cpp	2015-08-24 15:15:20.267413279 -0600
@@ -15,6 +15,7 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 #include "ghostpad.h"
+#include <glib.h>
 #include <gst/gstghostpad.h>
 
 namespace QGst {
diff -ur qtgstreamer-orig/git-r0/git/src/QGst/pad.cpp qtgstreamer/git-r0/git/src/QGst/pad.cpp
--- a/src/QGst/pad.cpp	2015-08-24 15:12:24.591417081 -0600
+++ b/src/QGst/pad.cpp	2015-08-24 15:14:55.751413810 -0600
@@ -20,6 +20,7 @@
 #include "query.h"
 #include "event.h"
 #include <QtCore/QDebug>
+#include <glib.h>
 #include <gst/gstpad.h>
 #include <gst/gstutils.h>
 
diff -ur qtgstreamer-orig/git-r0/git/src/QGst/taglist.cpp qtgstreamer/git-r0/git/src/QGst/taglist.cpp
--- a/src/QGst/taglist.cpp	2015-08-24 15:12:24.595417081 -0600
+++ b/src/QGst/taglist.cpp	2015-08-24 15:15:40.559412839 -0600
@@ -20,6 +20,10 @@
 #include "taglist.h"
 #include "sample.h"
 #include "../QGlib/string_p.h"
+#include "../QGlib/type.h"
+#include <glib.h>
+#include <glib-object.h>
+#include <gst/gstconfig.h>
 #include <gst/gsttaglist.h>
 #include <gst/gstvalue.h>
 #include <QtCore/QDebug>
