diff -Naur requests-2.7.0/requests/api.py requests-2.7.0_b/requests/api.py
--- requests-2.7.0/requests/api.py	2015-05-03 08:01:44.000000000 -0600
+++ requests-2.7.0_b/requests/api.py	2019-11-21 14:42:53.481331075 -0700
@@ -46,13 +46,11 @@
       <Response [200]>
     """
 
-    session = sessions.Session()
-    response = session.request(method=method, url=url, **kwargs)
-    # By explicitly closing the session, we avoid leaving sockets open which
-    # can trigger a ResourceWarning in some cases, and look like a memory leak
-    # in others.
-    session.close()
-    return response
+    # By using the 'with' statement we are sure the session is closed, thus we
+    # avoid leaving sockets open which can trigger a ResourceWarning in some
+    # cases, and look like a memory leak in others.
+    with sessions.Session() as session:
+        return session.request(method=method, url=url, **kwargs)
 
 
 def get(url, params=None, **kwargs):
