From 162840f8e5f2183a1d36772f3b5c6e45408c5801 Mon Sep 17 00:00:00 2001
From: Rakesh R <reddyrakesh926@gmail.com>
Date: Mon, 5 Aug 2019 11:47:04 -0600
Subject: [PATCH] Fix Implement stream group id.

---
 gst/mpegtsmux/mpegtsmux.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/gst/mpegtsmux/mpegtsmux.c b/gst/mpegtsmux/mpegtsmux.c
index 99562fd45..79d8e2eaa 100644
--- a/gst/mpegtsmux/mpegtsmux.c
+++ b/gst/mpegtsmux/mpegtsmux.c
@@ -1929,6 +1929,7 @@ mpegtsmux_prepare_srcpad (MpegTsMux * mux)
   GstSegment seg;
   /* we are not going to seek */
   GstEvent *new_seg;
+  GstEvent *stream_start;
   gchar s_id[32];
   GstCaps *caps = gst_caps_new_simple ("video/mpegts",
       "systemstream", G_TYPE_BOOLEAN, TRUE,
@@ -1938,9 +1939,11 @@ mpegtsmux_prepare_srcpad (MpegTsMux * mux)
 
   /* stream-start (FIXME: create id based on input ids) */
   g_snprintf (s_id, sizeof (s_id), "mpegtsmux-%08x", g_random_int ());
-  gst_pad_push_event (mux->srcpad, gst_event_new_stream_start (s_id));
+  stream_start = gst_event_new_stream_start (s_id);
+  gst_event_set_group_id (stream_start, gst_util_group_id_next ());
+  gst_pad_push_event (mux->srcpad, stream_start);
 
-  gst_segment_init (&seg, GST_FORMAT_TIME);
+gst_segment_init (&seg, GST_FORMAT_TIME);
   new_seg = gst_event_new_segment (&seg);
 
   /* Set caps on src pad from our template and push new segment */
-- 
2.17.1

