diff -rupN gst-plugins-bad-1.12.2/configure.ac gst-plugins-bad/configure.ac
--- gst-plugins-bad-1.12.2/configure.ac	2018-11-27 13:26:36.626275875 -0700
+++ gst-plugins-bad/configure.ac	2018-11-27 13:39:04.650781354 -0700
@@ -2939,7 +2939,7 @@ AG_GST_CHECK_FEATURE(QT, [Qt elements],
       else
         HAVE_QT="yes"
         HAVE_QT_WINDOWING="no"
-        QT_VERSION="`$PKG_CONFIG --define-prefix --modversion Qt5Core`"
+        QT_VERSION="`$PKG_CONFIG --modversion Qt5Core`"
         QPA_INCLUDE_PATH=`$PKG_CONFIG --variable=includedir Qt5Core`/QtGui/${QT_VERSION}/QtGui
         AS_IF([test -f "$PKG_CONFIG_SYSROOT_DIR/$QPA_INCLUDE_PATH/qpa/qplatformnativeinterface.h"], [
           QT_CFLAGS="$QT_CFLAGS -I$PKG_CONFIG_SYSROOT_DIR/$QPA_INCLUDE_PATH"
Binary files gst-plugins-bad-1.12.2/.configure.ac.swp and gst-plugins-bad/.configure.ac.swp differ
