--- numpy-1.7.0/numpy/distutils/ccompiler.py	2013-02-09 14:04:55.000000000 -0700
+++ numpy-1.7.0.b/numpy/distutils/ccompiler.py	2015-09-08 14:58:01.000000000 -0600
@@ -571,6 +571,12 @@
                                    runtime_library_dirs, libraries)
     lib_opts = []
     for i in r:
+        if "-L=" in i:
+            if "-L=/usr/lib" in i:
+                continue
+            else:
+                i = i.replace("-L=", "-L")
+
         if is_sequence(i):
             lib_opts.extend(list(i))
         else:
