--- openvpn-2.3.2/sample/sample-config-files/server.conf.org	2014-04-24 12:48:48.485756898 -0600
+++ openvpn-2.3.2/sample/sample-config-files/server.conf	2014-04-24 12:48:19.085756145 -0600
@@ -32,8 +32,8 @@
 port 1194
 
 # TCP or UDP server?
-;proto tcp
-proto udp
+proto tcp-server
+;proto udp
 
 # "dev tun" will create a routed IP tunnel,
 # "dev tap" will create an ethernet tunnel.
@@ -93,14 +93,14 @@
 # Each client will be able to reach the server
 # on 10.8.0.1. Comment this line out if you are
 # ethernet bridging. See the man page for more info.
-server 10.8.0.0 255.255.255.0
+server 172.29.0.0 255.255.255.0
 
 # Maintain a record of client <-> virtual IP address
 # associations in this file.  If OpenVPN goes down or
 # is restarted, reconnecting clients can be assigned
 # the same virtual IP address from the pool that was
 # previously assigned.
-ifconfig-pool-persist ipp.txt
+ifconfig-pool-persist /media/extra/conf/openvpn_server/ipp.txt
 
 # Configure server mode for ethernet bridging.
 # You must first use your OS's bridging capability
@@ -253,7 +253,7 @@
 
 # The maximum number of concurrently connected
 # clients we want to allow.
-;max-clients 100
+max-clients 50
 
 # It's a good idea to reduce the OpenVPN
 # daemon's privileges after initialization.
@@ -273,7 +273,7 @@
 # Output a short status file showing
 # current connections, truncated
 # and rewritten every minute.
-status openvpn-status.log
+status /media/extra/conf/openvpn_server/openvpn-status.log
 
 # By default, log messages will go to the syslog (or
 # on Windows, if running as a service, they will go to
@@ -283,7 +283,7 @@
 # while "log-append" will append to it.  Use one
 # or the other (but not both).
 ;log         openvpn.log
-;log-append  openvpn.log
+log-append  /media/extra/log/openvpnServer.log
 
 # Set the appropriate level of log
 # file verbosity.
@@ -297,4 +297,4 @@
 # Silence repeating messages.  At most 20
 # sequential messages of the same message
 # category will be output to the log.
-;mute 20
+mute 20
