diff -rupN openssl-1.0.2j/openssl.ld openssl-1.0.2j_new/openssl.ld
--- openssl-1.0.2j/openssl.ld	2017-08-28 15:37:48.737099367 -0600
+++ openssl-1.0.2j_new/openssl.ld	2017-08-28 15:37:16.077098545 -0600
@@ -207,6 +207,7 @@ OPENSSL_1.0.2d {
 		SSL_set_connect_state;
 		SSL_set_ex_data;
 		SSL_set_fd;
+        SSL_set_fd_rcv_msg;
 		SSL_set_generate_session_id;
 		SSL_set_info_callback;
 		SSL_set_msg_callback;
@@ -222,6 +223,7 @@ OPENSSL_1.0.2d {
 		SSL_set_session_ticket_ext;
 		SSL_set_session_ticket_ext_cb;
 		SSL_set_shutdown;
+        SSL_unset_shutdown;
 		SSL_set_SSL_CTX;
 		SSL_set_ssl_method;
 		SSL_set_tmp_dh_callback;
