diff -rupN gst1.0-fsl-plugins-4.0.3/ext-includes/fsl_media_types.h gst1.0-fsl-plugins/ext-includes/fsl_media_types.h
--- gst1.0-fsl-plugins-4.0.3/ext-includes/fsl_media_types.h     2015-03-02 20:49:48.000000000 -0700
+++ gst1.0-fsl-plugins/ext-includes/fsl_media_types.h   2019-09-03 09:03:57.413266968 -0600
@@ -101,6 +101,7 @@ typedef enum
     AUIDO_SPEEX,
     AUDIO_REAL, /* Real audio types */
     AUDIO_EC3,
+    AUDIO_OPUS,
 }AudioCodecType;


diff -rupN gst1.0-fsl-plugins-4.0.3/plugins/aiurdemux/aiurdemux.c gst1.0-fsl-plugins/plugins/aiurdemux/aiurdemux.c
--- gst1.0-fsl-plugins-4.0.3/plugins/aiurdemux/aiurdemux.c      2019-09-03 08:59:09.904774947 -0600
+++ gst1.0-fsl-plugins/plugins/aiurdemux/aiurdemux.c    2019-09-03 09:04:51.693606226 -0600
@@ -2371,6 +2371,14 @@ static void aiurdemux_parse_audio (GstAi
             stream->info.audio.n_channels, stream->info.audio.rate,
             stream->bitrate);
         break;
+      case AUDIO_OPUS:
+        codec = "OPUS";
+        mime =
+            g_strdup_printf
+            ("audio/x-opus, channel-mapping-family=(int)0, channels=(int)%ld, rate=(int)%ld, bitrate=(int)%ld",
+            stream->info.audio.n_channels, stream->info.audio.rate,
+            stream->bitrate);
+        break;
        default:
         break;
     }

