From 5ce6abbab8fa8e900224a6dd21c4984407e30630 Mon Sep 17 00:00:00 2001
From: Tanner Oakes <tanner.oakes@vivint.com>
Date: Wed, 28 Jun 2023 14:04:51 -0600
Subject: [PATCH] Inject keydata into openRTSP

---
 testProgs/playCommon.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/testProgs/playCommon.cpp b/testProgs/playCommon.cpp
index 326d500..b0326eb 100644
--- a/testProgs/playCommon.cpp
+++ b/testProgs/playCommon.cpp
@@ -103,6 +103,7 @@ unsigned short desiredPortNum = 0;
 portNumBits tunnelOverHTTPPortNum = 0;
 char* username = NULL;
 char* password = NULL;
+char* keyData = NULL;
 char* proxyServerName = NULL;
 unsigned short proxyServerPortNum = 0;
 unsigned char desiredAudioRTPPayloadFormat = 0;
@@ -142,6 +143,7 @@ void usage() {
        << "]" << (supportCodecSelection ? " [-A <audio-codec-rtp-payload-format-code>|-M <mime-subtype-name>]" : "")
        << " [-s <initial-seek-time>]|[-U <absolute-seek-time>] [-E <absolute-seek-end-time>] [-z <scale>] [-g user-agent]"
        << " [-k <username-for-REGISTER> <password-for-REGISTER>]"
+       << " [-e <key-salt>]"
        << " [-P <interval-in-seconds>] [-K]"
        << " [-w <width> -h <height>] [-f <frames-per-second>] [-y] [-H] [-Q [<measurement-interval>]] [-F <filename-prefix>] [-b <file-sink-buffer-size>] [-B <input-socket-buffer-size>] [-I <input-interface-ip-address>] [-m] [<url>|-R [<port-num>]] (or " << progName << " -o [-V] <url>)\n";
   shutdown();
@@ -544,6 +546,12 @@ int main(int argc, char** argv) {
       break;
     }
 
+    case 'e': { // specify a key to use with encrypted stream
+      keyData = argv[2];
+      ++argv; --argc;
+      break;
+    }
+
     default: {
       *env << "Invalid option: " << opt << "\n";
       usage();
@@ -707,7 +715,7 @@ void continueAfterDESCRIBE(RTSPClient*, int resultCode, char* resultString) {
   *env << "Opened URL \"" << streamURL << "\", returning a SDP description:\n" << sdpDescription << "\n";
 
   // Create a media session object from this SDP description:
-  session = MediaSession::createNew(*env, sdpDescription);
+  session = MediaSession::createNew(*env, sdpDescription, keyData);
   delete[] sdpDescription;
   if (session == NULL) {
     *env << "Failed to create a MediaSession object from the SDP description: " << env->getResultMsg() << "\n";
-- 
2.41.0

