diff --git a/code/framework/taurine/qt_helpers/qml_view.py b/code/framework/taurine/qt_helpers/qml_view.py
index 72a6850..ca5a66b 100644
--- a/code/framework/taurine/qt_helpers/qml_view.py
+++ b/code/framework/taurine/qt_helpers/qml_view.py
@@ -12,9 +12,9 @@ class QmlView(QtDeclarative.QDeclarativeView):
     def __init__(self):
         super().__init__()
 
-        if self._check_if_system_runs_on_target():
-            self._set_opengl_viewport()
-            self._set_optimizations()
+        #if self._check_if_system_runs_on_target():
+        #    self._set_opengl_viewport()
+        #    self._set_optimizations()
 
         self.setViewportUpdateMode(QtDeclarative.QDeclarativeView.FullViewportUpdate)
 
@@ -43,4 +43,4 @@ class QmlView(QtDeclarative.QDeclarativeView):
         self.viewport().setAttribute(Qt.Qt.WA_NoSystemBackground)
 
     def closeEvent(self, event):
-        sys.exit()
\ No newline at end of file
+        sys.exit()
diff --git a/code/roubaix/roubaix.qml b/code/roubaix/roubaix.qml
index 63172d8..d436a99 100644
--- a/code/roubaix/roubaix.qml
+++ b/code/roubaix/roubaix.qml
@@ -17,8 +17,8 @@ import "scripts/TouchlinkMsgLookup.js" as TouchlinkMsgLookup
 
 Item {
     id: rootWindow
-    width: 800
-    height: 480
+    width: 1024
+    height: 600
 
     property bool isSundanceConnected: sundance_connected
     property bool isIodConnected: iod_connected
@@ -41,7 +41,7 @@ Item {
         id: gBackground
         source: {
             if (showSplashBackground) {
-                return FindImage("splash_800x480.png")
+                return FindImage("splash_1024x600.png")
             } else {
                 if (gModel.systemTime.timeOfDay == "Morning")
                     return FindImage("background_morning_blurry.png")
diff --git a/code/roubaix/roubaix_conf.json b/code/roubaix/roubaix_conf.json
index 8e7373c..3087218 100644
--- a/code/roubaix/roubaix_conf.json
+++ b/code/roubaix/roubaix_conf.json
@@ -19,7 +19,7 @@
     {
         "host": "localhost",
         "port": 7140,
-        "connect": true
+        "connect": false
     },
 
     "Video":
@@ -33,35 +33,35 @@
     {
         "host": "localhost",
         "port": 7130,
-        "connect": true
+        "connect": false
     },
 
     "IOD":
     {
         "host": "localhost",
         "port": 7150,
-        "connect": true
+        "connect": false
     },
 
     "Network":
     {
         "host": "localhost",
         "port": 7170,
-        "connect": true
+        "connect": false
     },
 
     "Procman":
     {
         "host": "localhost",
         "port": 7016,
-        "connect": true
+        "connect": false
     },
 
     "Modem":
     {
         "host": "localhost",
         "port": 7190,
-        "connect": true
+        "connect": false
     },
 
     "Singletons":
diff --git a/code/roubaix/views/SplashView.qml b/code/roubaix/views/SplashView.qml
index 09812e2..deac09b 100644
--- a/code/roubaix/views/SplashView.qml
+++ b/code/roubaix/views/SplashView.qml
@@ -35,13 +35,13 @@ Views.BaseView {
         return ''
     }
 
-    width: 800
-    height: 480
+    width: 1024
+    height: 600
 
     // splash screen
     Image {
         anchors.fill: parent
-        source: FindImage("splash_800x480.png")
+        source: FindImage("splash_1024x600.png")
     }
 
     ListModel {
diff --git a/config/init.d/roubaix b/config/init.d/roubaix
index d7a8251..176eabd 100755
--- a/config/init.d/roubaix
+++ b/config/init.d/roubaix
@@ -15,7 +15,7 @@ then
 
     export TSLIB_TSDEVICE=/dev/input/event0                                                 
     export QWS_MOUSE_PROTO="Tslib:/dev/input/event0"                   
-    export QWS_DISPLAY=powervr                   
+    #export QWS_DISPLAY=powervr                   
 fi 
 
 # Turns on the Gstreamer debug message, they will show up in the roubaix log
