diff -rupN iotivity-2.0.0/bridging/plugins/lyric_plugin/honeywell_objects/honeywellThermostat.h iotivity/bridging/plugins/lyric_plugin/honeywell_objects/honeywellThermostat.h
--- iotivity-2.0.0/bridging/plugins/lyric_plugin/honeywell_objects/honeywellThermostat.h	2020-10-27 08:59:02.753861813 -0600
+++ iotivity/bridging/plugins/lyric_plugin/honeywell_objects/honeywellThermostat.h	2020-10-27 08:41:49.321135407 -0600
@@ -69,6 +69,7 @@ struct THERMOSTAT
 {
     THERMOSTAT(); // default constructor
     THERMOSTAT(const THERMOSTAT &data); // copy constructor
+    THERMOSTAT &operator=(const THERMOSTAT &data) = default; // copy assignment
 
     DEVICE_INFO devInfo;           // Device info
     bool isOnline;                 // Is thermostat online now?
diff -rupN iotivity-2.0.0/examples/OICMiddle/Client.cpp iotivity/examples/OICMiddle/Client.cpp
--- iotivity-2.0.0/examples/OICMiddle/Client.cpp	2020-10-27 08:59:02.753861813 -0600
+++ iotivity/examples/OICMiddle/Client.cpp	2020-10-27 08:41:49.321135407 -0600
@@ -51,7 +51,7 @@ void MiddleClient::foundOCResource(share
 
     try {
         wres = m_resourceMap.at(resourceID);
-    } catch (const std::out_of_range) {
+    } catch (const std::out_of_range&) {
         wres = new WrapResource(resourceID, resource);
         m_resourceMap[resourceID] = wres;
     }
@@ -85,7 +85,7 @@ void MiddleClient::addResource(WrapResou
     string resourceID = wres->getResourceID();
     try {
         m_resourceMap[resourceID];
-    } catch (const std::out_of_range) {
+    } catch (const std::out_of_range&) {
         m_resourceMap[resourceID] = wres;
     }
 }
diff -rupN iotivity-2.0.0/extlibs/hippomocks/hippomocks/HippoMocks/hippomocks.h /home/workdir/iotivity/extlibs/hippomocks/hippomocks/HippoMocks/hippomocks.h
--- iotivity-2.0.0/extlibs/hippomocks/hippomocks/HippoMocks/hippomocks.h	2020-09-11 11:01:33.141156139 -0600
+++ /home/workdir/iotivity/extlibs/hippomocks/hippomocks/HippoMocks/hippomocks.h	2020-09-11 10:48:50.049740537 -0600
@@ -4248,7 +4248,7 @@ noexcept(false)
 			{
 				latentException->rethrow();
 			}
-			catch(BASE_EXCEPTION e)
+			catch(BASE_EXCEPTION & e)
 			{
 				printf("Latent exception masked!\nException:\n%s\n", e.what());
 			}
diff -rupN iotivity-2.0.0/resource/examples/garageserver.cpp iotivity/resource/examples/garageserver.cpp
--- iotivity-2.0.0/resource/examples/garageserver.cpp	2020-10-27 08:59:02.753861813 -0600
+++ iotivity/resource/examples/garageserver.cpp	2020-10-27 08:41:49.334468870 -0600
@@ -388,7 +388,7 @@ int main(int /*argc*/, char** /*argv[1]*
         std::unique_lock<std::mutex> lock(blocker);
         cv.wait(lock);
     }
-    catch(OCException e)
+    catch(OCException &e)
     {
         oclog() << e.what();
     }
diff -rupN iotivity-2.0.0/resource/src/OCRepresentation.cpp iotivity/resource/src/OCRepresentation.cpp
--- iotivity-2.0.0/resource/src/OCRepresentation.cpp	2020-10-27 08:59:02.753861813 -0600
+++ iotivity/resource/src/OCRepresentation.cpp	2020-10-27 08:41:49.334468870 -0600
@@ -863,13 +863,13 @@ namespace OC
     OCRepresentation::AttributeItem OCRepresentation::operator[](const std::string& key)
     {
         OCRepresentation::AttributeItem attr{key, m_values};
-        return std::move(attr);
+        return attr;
     }
 
     const OCRepresentation::AttributeItem OCRepresentation::operator[](const std::string& key) const
     {
         OCRepresentation::AttributeItem attr{key, m_values};
-        return std::move(attr);
+        return attr;
     }
 
     const std::string& OCRepresentation::AttributeItem::attrname() const
@@ -1164,7 +1164,7 @@ namespace OC
         {
             to_string_visitor vis;
             boost::apply_visitor(vis, x->second);
-            return std::move(vis.str);
+            return vis.str;
         }
 
         return "";
@@ -1174,7 +1174,7 @@ namespace OC
     {
         to_string_visitor vis;
         boost::apply_visitor(vis, m_values[m_attrName]);
-        return std::move(vis.str);
+        return vis.str;
     }
 
     std::ostream& operator<<(std::ostream& os, const OCRepresentation::AttributeItem& ai)
diff -rupN iotivity-2.0.0/service/resource-container/examples/BMISensorBundle/src/inputSensors/WeightSensorApp/src/WeightSensorApp.cpp iotivity/service/resource-container/examples/BMISensorBundle/src/inputSensors/WeightSensorApp/src/WeightSensorApp.cpp
--- iotivity-2.0.0/service/resource-container/examples/BMISensorBundle/src/inputSensors/WeightSensorApp/src/WeightSensorApp.cpp	2020-10-27 08:59:02.753861813 -0600
+++ iotivity/service/resource-container/examples/BMISensorBundle/src/inputSensors/WeightSensorApp/src/WeightSensorApp.cpp	2020-10-27 08:41:49.334468870 -0600
@@ -221,7 +221,7 @@ int main()
 
         OC::OCPlatform::stopPresence();
     }
-    catch (std::exception e)
+    catch (std::exception& e)
     {
         cout << e.what() << endl;
     }
diff -rupN iotivity-2.0.0/service/resource-encapsulation/unittests/ResourceClientTest.cpp iotivity/service/resource-encapsulation/unittests/ResourceClientTest.cpp
--- iotivity-2.0.0/service/resource-encapsulation/unittests/ResourceClientTest.cpp	2020-10-27 08:59:01.997212717 -0600
+++ iotivity/service/resource-encapsulation/unittests/ResourceClientTest.cpp	2020-10-27 08:41:49.334468870 -0600
@@ -228,8 +228,8 @@ TEST_F(RemoteResourceObjectTest, QueryPa
         virtual ~CustomHandler() {}
     };
 
-    constexpr char PARAM_KEY[] { "aKey" };
-    constexpr char VALUE[] { "value" };
+    constexpr const char* PARAM_KEY = "aKey";
+    constexpr const char* VALUE = "value";
 
     auto mockHandler = mocks.Mock< CustomHandler >();
 
diff -rupN iotivity-2.0.0/service/scene-manager/sampleapp/linux/fanserver.cpp iotivity/service/scene-manager/sampleapp/linux/fanserver.cpp
--- iotivity-2.0.0/service/scene-manager/sampleapp/linux/fanserver.cpp	2020-10-27 08:59:01.997212717 -0600
+++ iotivity/service/scene-manager/sampleapp/linux/fanserver.cpp	2020-10-27 08:51:09.918955021 -0600
@@ -313,7 +313,7 @@ int main()
         std::unique_lock < std::mutex > lock(blocker);
         cv.wait(lock);
     }
-    catch (OCException e)
+    catch (OCException& e)
     {
         //log(e.what());
     }
