diff -rupN grpcio-1.25.0/setup.py grpcio/setup.py
--- grpcio-1.25.0/setup.py	2019-12-10 16:55:03.227109201 +0000
+++ grpcio/setup.py	2019-12-10 16:58:40.031791811 +0000
@@ -145,7 +145,7 @@ def check_linker_need_libatomic():
   """Test if linker on system needs libatomic."""
   code_test = (b'#include <atomic>\n' +
                b'int main() { return std::atomic<int64_t>{}; }')
-  cc_test = subprocess.Popen(['cc', '-x', 'c++', '-std=c++11', '-'],
+  cc_test = subprocess.Popen(['gcc', '-x', 'c++', '-std=c++11', '-'],
                              stdin=PIPE,
                              stdout=PIPE,
                              stderr=PIPE)
 
diff -rupN grpcio-1.25.0/src/core/lib/gpr/log_linux.cc grpcio/src/core/lib/gpr/log_linux.cc
--- grpcio-1.25.0/src/core/lib/gpr/log_linux.cc	2019-11-06 01:06:26.000000000 +0000
+++ grpcio/src/core/lib/gpr/log_linux.cc	2019-12-10 16:59:30.085432669 +0000
@@ -40,7 +40,13 @@
 #include <time.h>
 #include <unistd.h>
 
+#if  defined(__GLIBC__)
+#if !__GLIBC_PREREQ(2,29)
 static long sys_gettid(void) { return syscall(__NR_gettid); }
+#endif
+#else
+static long gettid(void) { return syscall(__NR_gettid); }
+#endif
 
 void gpr_log(const char* file, int line, gpr_log_severity severity,
              const char* format, ...) {
diff -rupN grpcio-1.25.0/src/python/grpcio/commands.py grpcio/src/python/grpcio/commands.py
--- grpcio-1.25.0/src/python/grpcio/commands.py	2019-12-10 16:55:15.490504289 +0000
+++ grpcio/src/python/grpcio/commands.py	2019-12-10 17:01:23.442779638 +0000
@@ -218,7 +218,7 @@ class BuildExt(build_ext.build_ext):
             when invoked in C mode. GCC is okay with this, while clang is not.
             """
             cc_test = subprocess.Popen(
-                ['cc', '-x', 'c', '-std=c++11', '-'],
+                ['gcc', '-x', 'c', '-std=c++11', '-'],
                 stdin=subprocess.PIPE,
                 stdout=subprocess.PIPE,
                 stderr=subprocess.PIPE)
