diff -rupN grpcio-1.31.0/third_party/boringssl-with-bazel/src/include/openssl/base.h grpcio/third_party/boringssl-with-bazel/src/include/openssl/base.h
--- grpcio-1.31.0/third_party/boringssl-with-bazel/src/include/openssl/base.h	2020-09-02 16:22:01.085947707 -0600
+++ grpcio/third_party/boringssl-with-bazel/src/include/openssl/base.h	2020-09-02 16:23:00.967144124 -0600
@@ -108,6 +108,9 @@ extern "C" {
 #elif defined(__mips__) && defined(__LP64__)
 #define OPENSSL_64_BIT
 #define OPENSSL_MIPS64
+#elif defined(__riscv) && defined(__riscv_xlen)
+#define OPENSSL_64_BIT
+#define OPENSSL_RISCV64
 #elif defined(__pnacl__)
 #define OPENSSL_32_BIT
 #define OPENSSL_PNACL
