diff --git a/setup.py b/setup.py
index 453c1c5..37976ea 100644
--- a/setup.py
+++ b/setup.py
@@ -24,9 +24,9 @@ from __future__ import absolute_import, division, print_function
 import os
 import sys
 
-from setuptools import setup
-from setuptools.command.build_py import build_py as _build_py
-from setuptools.command.sdist import sdist as _sdist
+from distutils.core import setup
+from distutils.command.build_py import build_py as _build_py
+from distutils.command.sdist import sdist as _sdist
 
 from flent.build_info import VERSION
 from glob import glob
