--- Python-3.3.2/setup.py	2013-09-24 12:52:10.000000000 -0600
+++ Python-3.3.2.b/setup.py	2013-09-24 12:56:59.418414693 -0600
@@ -1774,14 +1774,16 @@
                 if not self.verbose:
                     config_args.append("-q")
 
+                config_args += ['--host=%s' % os.environ["HOST_SYS"], ]
+			
                 # Pass empty CFLAGS because we'll just append the resulting
                 # CFLAGS to Python's; -g or -O2 is to be avoided.
-                cmd = "cd %s && env CFLAGS='' '%s/configure' %s" \
-                      % (ffi_builddir, ffi_srcdir, " ".join(config_args))
+                cmd = "(cd %s && autoconf -W cross)  && (cd %s && env CFLAGS='' '%s/configure' %s)" \
+                        % (ffi_srcdir, ffi_builddir, ffi_srcdir, " ".join(config_args))
 
                 res = os.system(cmd)
                 if res or not os.path.exists(ffi_configfile):
-                    print("Failed to configure _ctypes module")
+                    print ("Failed to configure _ctypes module (res=%d) or missing conffile=%s" % ( res, ffi_configfile ))
                     return False
 
             fficonfig = {}
