diff -r 441eb2efeb71 src/gsttools/qgstutils.cpp
--- a/src/gsttools/qgstutils.cpp	Fri Sep 18 03:09:13 2015 +0000
+++ b/src/gsttools/qgstutils.cpp	Fri Sep 18 03:11:36 2015 +0000
@@ -52,6 +52,48 @@
 QT_BEGIN_NAMESPACE
 
 //internal
+static void qt_extension_log_func (GstDebugCategory * category,
+    GstDebugLevel level, const gchar * file, const gchar * function,
+    gint line, GObject * object, GstDebugMessage * message, gpointer unused)
+{
+  const gchar *dbg_msg;
+  gint pid;
+  gchar *obj = NULL;
+  
+  if (level > gst_debug_category_get_threshold(category))
+    return;
+  
+  pid = getpid();
+  
+  //if (object) {
+  //  obj = gst_debug_print_object (object);
+  //} else {
+    obj = g_strdup ("");
+  //}
+  
+  dbg_msg = gst_debug_message_get (message);
+  //fail_unless (dbg_msg != NULL);
+
+  switch (level) {
+    case GST_LEVEL_INFO:
+    case GST_LEVEL_LOG:
+    case GST_LEVEL_DEBUG:
+      qDebug() << "GSTREAMER: " << pid << " " << g_thread_self() << " " << gst_debug_category_get_name(category) << " " << file << ":" << line <<" " << function << " " << obj << " "<< dbg_msg;
+      break;
+    case GST_LEVEL_WARNING:
+      qWarning()  << "GSTREAMER: " << pid << " " << g_thread_self() << " " << gst_debug_category_get_name(category) << " " << file << ":" << line <<" " << function << " " << obj << " "<< dbg_msg;
+      break;
+    case GST_LEVEL_ERROR:
+      qCritical("GSTREAMER: %d %s %s:%d %s %s %s",pid,gst_debug_category_get_name(category),file,line,function,obj,dbg_msg);
+      break;
+    default:
+      qCritical("GSTREAMER(%d): %d %s %s:%d %s %s %s",level,pid,gst_debug_category_get_name(category),file,line,function,obj,dbg_msg);
+  }
+
+  g_free(obj);
+  
+}
+
 static void addTagToMap(const GstTagList *list,
                         const gchar *tag,
                         gpointer user_data)
@@ -321,6 +363,8 @@
     if (!initialized) {
         initialized = true;
         gst_init(NULL, NULL);
+        gst_debug_remove_log_function (gst_debug_log_default);
+	gst_debug_add_log_function (qt_extension_log_func, NULL,NULL);
     }
 }
 
