diff -rupN qtmultimedia-opensource-src-5.4.1/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp qmultimedia/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
--- qtmultimedia-opensource-src-5.4.1/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp	2019-01-08 15:42:01.234279678 -0700
+++ qmultimedia/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp	2019-01-08 15:45:44.981219043 -0700
@@ -1672,8 +1672,8 @@ void QGstreamerPlayerSession::playbinNot
         qDebug() << "Current source is a non-live source";
 #endif
 
-    if (self->m_videoSink)
-        g_object_set(G_OBJECT(self->m_videoSink), "sync", !self->m_isLiveSource, NULL);
+    //if (self->m_videoSink)
+    //    g_object_set(G_OBJECT(self->m_videoSink), "sync", !self->m_isLiveSource, NULL);
 
     gst_object_unref(source);
 }
