diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/gsttools/gsttools.pro qtmultimedia-opensource-src-5.4.1/src/gsttools/gsttools.pro
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/gsttools/gsttools.pro	2019-10-09 14:50:51.086856082 -0600
+++ qtmultimedia-opensource-src-5.4.1/src/gsttools/gsttools.pro	2019-10-09 14:28:23.066212521 -0600
@@ -21,7 +21,8 @@ PKGCONFIG += \
     gstreamer-base-$$GST_VERSION \
     gstreamer-audio-$$GST_VERSION \
     gstreamer-video-$$GST_VERSION \
-    gstreamer-pbutils-$$GST_VERSION
+    gstreamer-pbutils-$$GST_VERSION \
+    gstreamer-fsl-1.0
 
 equals(GST_VERSION,"0.10") {
     PKGCONFIG_PRIVATE += gstreamer-interfaces-0.10
diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/gsttools/qgstvideobuffer.cpp qtmultimedia-opensource-src-5.4.1/src/gsttools/qgstvideobuffer.cpp
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/gsttools/qgstvideobuffer.cpp	2019-10-09 14:50:51.090189433 -0600
+++ qtmultimedia-opensource-src-5.4.1/src/gsttools/qgstvideobuffer.cpp	2019-10-09 14:42:10.564066752 -0600
@@ -33,6 +33,10 @@
 
 #include "qgstvideobuffer_p.h"
 
+extern "C" {
+#include <gstreamer-1.0/gst/imx-mm/gstallocatorphymem.h>
+}
+
 QT_BEGIN_NAMESPACE
 
 #if GST_CHECK_VERSION(1,0,0)
@@ -48,6 +52,11 @@ QGstVideoBuffer::QGstVideoBuffer(GstBuff
     , m_mode(NotMapped)
 {
     gst_buffer_ref(m_buffer);
+
+    PhyMemBlock *meta = gst_buffer_query_phymem_block(buffer);
+    if (meta && meta->paddr != 0) {
+	setMetaData("imx_phys_addr", (quint32)meta->paddr);
+    }
 }
 
 #if GST_CHECK_VERSION(1,0,0)
@@ -68,6 +77,12 @@ QGstVideoBuffer::QGstVideoBuffer(GstBuff
     , m_handle(handle)
 {
     gst_buffer_ref(m_buffer);
+
+    PhyMemBlock *meta = gst_buffer_query_phymem_block(buffer);
+    if (meta && meta->paddr != 0) {
+	setMetaData("imx_phys_addr", (quint32)meta->paddr);
+    }
+
 }
 
 QGstVideoBuffer::~QGstVideoBuffer()
diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/gsttools_headers/qgstvideobuffer_p.h qtmultimedia-opensource-src-5.4.1/src/multimedia/gsttools_headers/qgstvideobuffer_p.h
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/gsttools_headers/qgstvideobuffer_p.h	2019-10-09 14:50:51.096856134 -0600
+++ qtmultimedia-opensource-src-5.4.1/src/multimedia/gsttools_headers/qgstvideobuffer_p.h	2019-10-09 13:42:45.204988903 -0600
@@ -71,6 +71,7 @@ public:
 
     ~QGstVideoBuffer();
 
+    GstBuffer *buffer() const { return m_buffer; }
     MapMode mapMode() const;
 
 #if GST_CHECK_VERSION(1,0,0)
diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.cpp qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.cpp
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.cpp	2015-02-16 21:57:20.000000000 -0700
+++ qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.cpp	2019-10-09 14:33:07.817789937 -0600
@@ -263,6 +263,23 @@ QVariant QAbstractVideoBuffer::handle()
     return QVariant();
 }
 
+QVariantMap QAbstractVideoBuffer::availableMetaData() const
+{
+    return m_metadata;
+}
+
+QVariant QAbstractVideoBuffer::metaData(const QString &key) const
+{
+    return m_metadata.value(key);
+}
+
+void QAbstractVideoBuffer::setMetaData(const QString &key, const QVariant &value)
+{
+    if (!value.isNull())
+        m_metadata.insert(key, value);
+    else
+        m_metadata.remove(key);
+}
 
 int QAbstractPlanarVideoBufferPrivate::map(
         QAbstractVideoBuffer::MapMode mode, int *numBytes, int bytesPerLine[4], uchar *data[4])
diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.h qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.h
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.h	2015-02-16 21:57:20.000000000 -0700
+++ qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.h	2019-10-09 14:38:10.509438287 -0600
@@ -39,6 +39,9 @@
 
 
 #include <QtCore/qmetatype.h>
+#include <QVariantMap>
+
+
 
 QT_BEGIN_NAMESPACE
 
@@ -83,11 +86,16 @@ public:
 
     virtual QVariant handle() const;
 
+    QVariantMap availableMetaData() const;
+    QVariant metaData(const QString &key) const;
+    void setMetaData(const QString &key, const QVariant &value);
+
 protected:
     QAbstractVideoBuffer(QAbstractVideoBufferPrivate &dd, HandleType type);
 
     QAbstractVideoBufferPrivate *d_ptr;  // For expansion, not used currently
     HandleType m_type;
+    QVariantMap m_metadata;
 
 private:
     Q_DECLARE_PRIVATE(QAbstractVideoBuffer)
diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qvideoframe.cpp qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qvideoframe.cpp
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qvideoframe.cpp	2019-10-09 14:50:51.346857471 -0600
+++ qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qvideoframe.cpp	2019-10-09 14:35:12.558471906 -0600
@@ -309,6 +309,13 @@ QVideoFrame::QVideoFrame(
     : d(new QVideoFramePrivate(size, format))
 {
     d->buffer = buffer;
+
+    if (buffer) {
+        for (auto it = buffer->availableMetaData().cbegin(),
+             end = buffer->availableMetaData().cend(); it != end; ++it) {
+            d->metadata.insert(it.key(), it.value());
+        }
+    }
 }
 
 /*!
diff -rupN ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp qtmultimedia-opensource-src-5.4.1/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp
--- ../5.4.1-r12/qtmultimedia-opensource-src-5.4.1/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp	2019-10-09 14:50:51.300190554 -0600
+++ qtmultimedia-opensource-src-5.4.1/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp	2019-10-09 14:43:58.784648331 -0600
@@ -165,12 +165,16 @@ GLuint QSGVivanteVideoMaterial::vivanteM
             const uchar *constBits = vF.bits();
             void *bits = (void*)constBits;
 
+            GLuint physical = ~0U;
+	    QVariant p = vF.metaData("imx_phys_addr");
+            if (p.isValid()) {
+                physical = p.toUInt();
+            }
+
 #ifdef QT_VIVANTE_VIDEO_DEBUG
             qDebug() << Q_FUNC_INFO << "new texture, texId: " << tmpTexId << "; constBits: " << constBits;
 #endif
 
-            GLuint physical = ~0U;
-
             glBindTexture(GL_TEXTURE_2D, tmpTexId);
             glTexDirectVIVMap_LOCAL(GL_TEXTURE_2D,
                                      vF.width(), vF.height() + vF.extraHeight(),
