diff -rupN qtmultimedia-opensource-src-5.4.1/src/gsttools/gsttools.pro qtmultimedia/src/gsttools/gsttools.pro
--- qtmultimedia-opensource-src-5.4.1/src/gsttools/gsttools.pro	2019-12-20 14:45:58.198242158 -0700
+++ qtmultimedia/src/gsttools/gsttools.pro	2019-12-20 14:47:00.025240163 -0700
@@ -31,6 +31,14 @@ config_resourcepolicy {
     PKGCONFIG_PRIVATE += libresourceqt5
 }
 
+config_gpu_vivante {
+    packagesExist(gstimxcommon) {
+        message(Enabling GStreamer IMX extensions)
+        DEFINES += GST_IMX_EXTENSIONS
+        PKGCONFIG += gstimxcommon
+    }
+}
+
 # Header files must go inside source directory of a module
 # to be installed by syncqt.
 INCLUDEPATH += ../multimedia/gsttools_headers/
diff -rupN qtmultimedia-opensource-src-5.4.1/src/gsttools/qgstvideobuffer.cpp qtmultimedia/src/gsttools/qgstvideobuffer.cpp
--- qtmultimedia-opensource-src-5.4.1/src/gsttools/qgstvideobuffer.cpp	2019-12-20 14:45:58.201575509 -0700
+++ qtmultimedia/src/gsttools/qgstvideobuffer.cpp	2019-12-20 14:48:14.162304721 -0700
@@ -33,6 +33,10 @@
 
 #include "qgstvideobuffer_p.h"
 
+#ifdef GST_IMX_EXTENSIONS
+#include <gstreamer-1.0/gst/allocators/imx/phys_mem_meta.h>
+#endif
+
 QT_BEGIN_NAMESPACE
 
 QGstVideoBuffer::QGstVideoBuffer(GstBuffer *buffer, int bytesPerLine)
@@ -42,6 +46,12 @@ QGstVideoBuffer::QGstVideoBuffer(GstBuff
     , m_mode(NotMapped)
 {
     gst_buffer_ref(m_buffer);
+#ifdef GST_IMX_EXTENSIONS
+    GstImxPhysMemMeta *meta = GST_IMX_PHYS_MEM_META_GET(buffer);
+    if (meta && meta->phys_addr != 0) {
+        setMetaData("imx_phys_addr", (quint32)meta->phys_addr);
+    }
+#endif
 }
 
 QGstVideoBuffer::QGstVideoBuffer(GstBuffer *buffer, int bytesPerLine,
@@ -54,6 +64,12 @@ QGstVideoBuffer::QGstVideoBuffer(GstBuff
     , m_handle(handle)
 {
     gst_buffer_ref(m_buffer);
+#ifdef GST_IMX_EXTENSIONS
+    GstImxPhysMemMeta *meta = GST_IMX_PHYS_MEM_META_GET(buffer);
+    if (meta && meta->phys_addr != 0) {
+        setMetaData("imx_phys_addr", (quint32)meta->phys_addr);
+    }
+#endif
 }
 
 QGstVideoBuffer::~QGstVideoBuffer()
diff -rupN qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.cpp qtmultimedia/src/multimedia/video/qabstractvideobuffer.cpp
--- qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.cpp	2019-12-20 14:45:57.841573581 -0700
+++ qtmultimedia/src/multimedia/video/qabstractvideobuffer.cpp	2019-12-20 14:48:44.412467237 -0700
@@ -263,6 +263,23 @@ QVariant QAbstractVideoBuffer::handle()
     return QVariant();
 }
 
+QVariantMap QAbstractVideoBuffer::availableMetaData() const
+{
+    return m_metadata;
+}
+
+QVariant QAbstractVideoBuffer::metaData(const QString &key) const
+{
+    return m_metadata.value(key);
+}
+
+void QAbstractVideoBuffer::setMetaData(const QString &key, const QVariant &value)
+{
+    if (!value.isNull())
+        m_metadata.insert(key, value);
+    else
+        m_metadata.remove(key);
+}
 
 int QAbstractPlanarVideoBufferPrivate::map(
         QAbstractVideoBuffer::MapMode mode, int *numBytes, int bytesPerLine[4], uchar *data[4])
diff -rupN qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.h qtmultimedia/src/multimedia/video/qabstractvideobuffer.h
--- qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qabstractvideobuffer.h	2019-12-20 14:45:57.841573581 -0700
+++ qtmultimedia/src/multimedia/video/qabstractvideobuffer.h	2019-12-20 14:49:42.126110875 -0700
@@ -39,6 +39,7 @@
 
 
 #include <QtCore/qmetatype.h>
+#include <QVariantMap>
 
 QT_BEGIN_NAMESPACE
 
@@ -82,12 +83,16 @@ public:
     virtual void unmap() = 0;
 
     virtual QVariant handle() const;
+    QVariantMap availableMetaData() const;
+    QVariant metaData(const QString &key) const;
+    void setMetaData(const QString &key, const QVariant &value);
 
 protected:
     QAbstractVideoBuffer(QAbstractVideoBufferPrivate &dd, HandleType type);
 
     QAbstractVideoBufferPrivate *d_ptr;  // For expansion, not used currently
     HandleType m_type;
+    QVariantMap m_metadata;
 
 private:
     Q_DECLARE_PRIVATE(QAbstractVideoBuffer)
diff -rupN qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qvideoframe.cpp qtmultimedia/src/multimedia/video/qvideoframe.cpp
--- qtmultimedia-opensource-src-5.4.1/src/multimedia/video/qvideoframe.cpp	2019-12-20 14:45:58.484910360 -0700
+++ qtmultimedia/src/multimedia/video/qvideoframe.cpp	2019-12-20 14:50:04.026228691 -0700
@@ -309,6 +309,13 @@ QVideoFrame::QVideoFrame(
     : d(new QVideoFramePrivate(size, format))
 {
     d->buffer = buffer;
+
+    if (buffer) {
+        for (auto it = buffer->availableMetaData().cbegin(),
+             end = buffer->availableMetaData().cend(); it != end; ++it) {
+            d->metadata.insert(it.key(), it.value());
+        }
+    }
 }
 
 /*!
diff -rupN qtmultimedia-opensource-src-5.4.1/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp qtmultimedia/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp
--- qtmultimedia-opensource-src-5.4.1/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp	2019-12-20 14:45:58.428243390 -0700
+++ qtmultimedia/src/plugins/videonode/imx6/qsgvivantevideomaterial.cpp	2019-12-20 14:50:39.966422112 -0700
@@ -165,12 +165,15 @@ GLuint QSGVivanteVideoMaterial::vivanteM
             const uchar *constBits = vF.bits();
             void *bits = (void*)constBits;
 
+            GLuint physical = ~0U;
+            QVariant p = vF.metaData("imx_phys_addr");
+            if (p.isValid()) {
+                physical = p.toUInt();
+            }
+
 #ifdef QT_VIVANTE_VIDEO_DEBUG
             qDebug() << Q_FUNC_INFO << "new texture, texId: " << tmpTexId << "; constBits: " << constBits;
 #endif
-
-            GLuint physical = ~0U;
-
             glBindTexture(GL_TEXTURE_2D, tmpTexId);
             glTexDirectVIVMap_LOCAL(GL_TEXTURE_2D,
                                      vF.width(), vF.height() + vF.extraHeight(),
