diff -rupN qtmultimedia-opensource-src-5.4.1/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp qtmultimedia/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp
--- qtmultimedia-opensource-src-5.4.1/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp	2019-01-08 15:42:01.234279678 -0700
+++ qtmultimedia/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.cpp	2019-01-08 15:49:49.013547531 -0700
@@ -129,6 +129,7 @@ QGstreamerPlayerSession::QGstreamerPlaye
      m_videoProbe(0),
      m_audioProbe(0),
      m_volume(100),
+     m_latency(200),
      m_playbackRate(1.0),
      m_muted(false),
      m_audioAvailable(false),
@@ -367,6 +368,7 @@ void QGstreamerPlayerSession::loadFromUr
         emit tagsChanged();
 
         g_object_set(G_OBJECT(m_playbin), "uri", m_request.url().toEncoded().constData(), NULL);
+	g_signal_connect(G_OBJECT(m_playbin), "source-setup", G_CALLBACK(handleSourceCreated), this);
 
         if (!m_streamTypes.isEmpty()) {
             m_streamProperties.clear();
@@ -1788,6 +1790,15 @@ GstAutoplugSelectResult QGstreamerPlayer
     return res;
 }
 
+void QGstreamerPlayerSession::handleSourceCreated(GstElement *pipe, GstElement *source, QGstreamerPlayerSession *session)
+{
+    Q_UNUSED(pipe);
+
+    if (session->m_isLiveSource)
+        g_object_set(source, "latency", session->m_latency, NULL);
+}
+
+
 void QGstreamerPlayerSession::handleElementAdded(GstBin *bin, GstElement *element, QGstreamerPlayerSession *session)
 {
     Q_UNUSED(bin);
diff -rupN qtmultimedia-opensource-src-5.4.1/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.h qtmultimedia/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.h
--- qtmultimedia-opensource-src-5.4.1/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.h	2019-01-08 15:42:00.581278011 -0700
+++ qtmultimedia/src/plugins/gstreamer/mediaplayer/qgstreamerplayersession.h	2019-01-08 15:50:34.482795058 -0700
@@ -173,6 +173,7 @@ private:
 #if !GST_CHECK_VERSION(1,0,0)
     static void insertColorSpaceElement(GstElement *element, gpointer data);
 #endif
+    static void handleSourceCreated(GstElement *pipe, GstElement *source, QGstreamerPlayerSession *session);
     static void handleElementAdded(GstBin *bin, GstElement *element, QGstreamerPlayerSession *session);
     static void handleStreamsChange(GstBin *bin, gpointer user_data);
     static GstAutoplugSelectResult handleAutoplugSelect(GstBin *bin, GstPad *pad, GstCaps *caps, GstElementFactory *factory, QGstreamerPlayerSession *session);
@@ -225,6 +226,7 @@ private:
     QGstreamerAudioProbeControl *m_audioProbe;
 
     int m_volume;
+    int m_latency;
     qreal m_playbackRate;
     bool m_muted;
     bool m_audioAvailable;
