diff --git a/src/gsttools/qgstreamerplayersession.cpp b/src/gsttools/qgstreamerplayersession.cpp
index adf11b022..4bce1969f 100644
--- a/src/gsttools/qgstreamerplayersession.cpp
+++ b/src/gsttools/qgstreamerplayersession.cpp
@@ -617,6 +617,21 @@ bool QGstreamerPlayerSession::isAudioAvailable() const
     return m_audioAvailable;
 }
 
+#if GST_CHECK_VERSION(1,0,0)
+static GstPadProbeReturn drop_until_keyframe_cb(GstPad *pad, GstPadProbeInfo *info, gpointer user_data)
+{
+    Q_UNUSED(pad);
+    Q_UNUSED(user_data);
+    GstBuffer *buffer = GST_PAD_PROBE_INFO_BUFFER(info);
+    qWarning() << GST_BUFFER_FLAGS(buffer);
+    if (GST_BUFFER_FLAG_IS_SET (buffer, GST_BUFFER_FLAG_DELTA_UNIT)) {
+        return GST_PAD_PROBE_DROP;
+      }
+
+    return GST_PAD_PROBE_REMOVE;
+}
+#endif
+
 #if GST_CHECK_VERSION(1,0,0)
 static GstPadProbeReturn block_pad_cb(GstPad *pad, GstPadProbeInfo *info, gpointer user_data)
 #else
@@ -1867,6 +1882,13 @@ void QGstreamerPlayerSession::handleElementAdded(GstBin *bin, GstElement *elemen
         //Don't touch other bins since they may have unrelated queues
         g_signal_connect(element, "element-added",
                          G_CALLBACK(handleElementAdded), session);
+    } else if (g_str_has_prefix(elementName, "rtph264depay")) {
+        // Drop all frames until keyframe is detected
+#if GST_CHECK_VERSION(1,0,0)
+        GstPad *pad = gst_element_get_static_pad(element, "src");
+        gst_pad_add_probe(pad, (GstPadProbeType)(GST_PAD_PROBE_TYPE_BUFFER), drop_until_keyframe_cb, session, nullptr);
+        gst_object_unref(GST_OBJECT(pad));
+#endif
     }
 
     g_free(elementName);
