diff -ruN speedtest-cli.orig/speedtest.py speedtest-cli/speedtest.py
--- speedtest-cli-master/speedtest.py	2017-04-25 10:49:26.000000000 -0600
+++ speedtest-cli/speedtest.py	2017-06-16 16:29:44.973695197 -0600
@@ -1447,7 +1447,7 @@
     results = speedtest.results
 
     printer('Hosted by %(sponsor)s (%(name)s) [%(d)0.2f km]: '
-            '%(latency)s ms' % results.server, quiet)
+            '%(latency)0.2f ms' % results.server, quiet)
 
     if args.download:
         printer('Testing download speed', quiet,
@@ -1472,7 +1472,7 @@
         printer('Skipping upload test')
 
     if args.simple:
-        print_('Ping: %s ms\nDownload: %0.2f M%s/s\nUpload: %0.2f M%s/s' %
+        print_('Ping: %0.2f ms\nDownload: %0.2f M%s/s\nUpload: %0.2f M%s/s' %
                (results.ping,
                 (results.download / 1000.0 / 1000.0) / args.units[1],
                 args.units[0],

