From 57532f1331a642d2578f9f5ffbe43aa2f2e865a1 Mon Sep 17 00:00:00 2001
From: Stephen Farnsworth <robotoman@gmail.com>
Date: Mon, 13 Jun 2016 14:56:53 -0600
Subject: [PATCH] Modified setup.py to work with python3 for yocto

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 82cf121..aa04050 100755
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,5 @@
 #!/usr/bin/env python
-from setuptools import setup, find_packages
+from distutils.core import setup
 
 # versioneer config
 import versioneer
@@ -39,7 +39,7 @@ setup(
     author_email='adam@movq.us',
     url='https://github.com/ahknight/httpsig',
     license='MIT',
-    packages=find_packages(),
+    packages=['httpsig', 'httpsig.tests'],
     include_package_data=True,
     zip_safe=True,
     install_requires=['pycrypto', 'six'],
-- 
2.7.4

