diff --git a/httpsig/sign.py b/httpsig/sign.py
index 94e2180..7129212 100644
--- a/httpsig/sign.py
+++ b/httpsig/sign.py
@@ -28,7 +28,10 @@ class Signer(object):
 
         self._rsa = None
         self._hash = None
-        self.sign_algorithm, self.hash_algorithm = algorithm.split('-')
+        try:
+            self.sign_algorithm, self.hash_algorithm = algorithm.split('-')
+        except ValueError:
+            self.sign_algorithm = algorithm
 
         if self.sign_algorithm == 'rsa':
             try:
@@ -42,9 +45,15 @@ class Signer(object):
             self._hash = HMAC.new(secret,
                                   digestmod=HASHES[self.hash_algorithm])
 
+        elif self.sign_algorithm == 'ed25519':
+            self._ed25519 = base64.b64decode(secret)
+
     @property
     def algorithm(self):
-        return '%s-%s' % (self.sign_algorithm, self.hash_algorithm)
+        if self.sign_algorithm == 'ed25519':
+            return self.sign_algorithm
+        else:
+            return '%s-%s' % (self.sign_algorithm, self.hash_algorithm)
 
     def _sign_rsa(self, data):
         if isinstance(data, six.string_types):
@@ -60,6 +69,14 @@ class Signer(object):
         hmac.update(data)
         return hmac.digest()
 
+    def _sign_ed25519(self, data):
+        if isinstance(data, six.string_types): data = data.encode("ascii")
+        try:
+            from pysodium import crypto_sign_detached
+            return crypto_sign_detached(data, self._ed25519)
+        except ImportError:
+            raise HttpSigException("Unsupported algorithm")
+
     def sign(self, data):
         if isinstance(data, six.string_types):
             data = data.encode("ascii")
@@ -68,6 +85,8 @@ class Signer(object):
             signed = self._sign_rsa(data)
         elif self._hash:
             signed = self._sign_hmac(data)
+        elif self._ed25519:
+            signed = self._sign_ed25519(data)
         if not signed:
             raise SystemError('No valid encryptor found.')
         return base64.b64encode(signed).decode("ascii")
diff --git a/httpsig/tests/test_verify.py b/httpsig/tests/test_verify.py
index 6e6d9eb..4418cca 100755
--- a/httpsig/tests/test_verify.py
+++ b/httpsig/tests/test_verify.py
@@ -2,6 +2,7 @@
 import sys
 import os
 import unittest
+from base64 import b64encode
 
 from httpsig.sign import HeaderSigner, Signer
 from httpsig.verify import HeaderVerifier, Verifier
@@ -215,6 +216,22 @@ class TestVerifyRSASHA512(TestVerifyRSASHA1):
         super(TestVerifyRSASHA512, self).setUp()
         self.algorithm = "rsa-sha512"
 
+try:
+    from pysodium import crypto_sign_keypair
+    class TestVerifyED25519(TestVerifyHMACSHA1):
+        def setUp(self):
+            public_key, private_key = crypto_sign_keypair()
+            self.keyId = "Test"
+            self.algorithm = "ed25519"
+            self.sign_secret = b64encode(private_key)
+            self.verify_secret = b64encode(public_key)
+
+except ImportError, e:
+    if e.message.find('pysodium') >= 0:
+        class TestMissingDependency(unittest.TestCase):
+            @unittest.skip('Optional dependency missing - ' + e.message)
+            def test_ed25519():
+                pass
 
 class TestVerifyRSASHA512ChangeHeader(TestVerifyRSASHA1):
     sign_header = 'Signature'
diff --git a/httpsig/utils.py b/httpsig/utils.py
index 5f80ef0..1634638 100644
--- a/httpsig/utils.py
+++ b/httpsig/utils.py
@@ -19,7 +19,8 @@ ALGORITHMS = frozenset([
                 'rsa-sha512',
                 'hmac-sha1',
                 'hmac-sha256',
-                'hmac-sha512'])
+                'hmac-sha512',
+                'ed25519'])
 HASHES = {'sha1':   SHA,
           'sha256': SHA256,
           'sha512': SHA512}
diff --git a/httpsig/verify.py b/httpsig/verify.py
index 17e313d..c346fc9 100644
--- a/httpsig/verify.py
+++ b/httpsig/verify.py
@@ -37,6 +37,17 @@ class Verifier(Signer):
             s = base64.b64decode(signature)
             return ct_bytes_compare(h, s)
 
+        elif self.sign_algorithm == 'ed25519':
+            try:
+                from pysodium import crypto_sign_verify_detached
+                crypto_sign_verify_detached(b64decode(signature), data, self._ed25519)
+            except ValueError:
+                return False
+            except ImportError:
+                raise HttpSigException("Unsupported algorithm.")
+                return False
+            return True
+
         else:
             raise HttpSigException("Unsupported algorithm.")
