diff --git a/CMakeLists.txt b/CMakeLists.txt
index 749667f..1848327 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -464,28 +464,6 @@ endif()
 # Build the targets
 add_subdirectory(src)
 
-# Enable the testing of the library/executable
-include(CTest)
-if (CIVETWEB_BUILD_TESTING)
-  # Check unit testing framework Version
-  set(CIVETWEB_CHECK_VERSION 0.11.0 CACHE STRING
-    "The version of Check unit testing framework to build and include statically")
-  set_property(CACHE CIVETWEB_CHECK_VERSION PROPERTY VALUE ${CIVETWEB_CHECK_VERSION})
-  message(STATUS "Check Unit Testing Framework Version - ${CIVETWEB_CHECK_VERSION}")
-  mark_as_advanced(CIVETWEB_CHECK_VERSION)
-
-  # Check unit testing framework Verification Hash
-  # Hash for Check 0.10.0: 67a34c40b5bc888737f4e5ae82e9939f
-  # Hash for Check 0.11.0: 1b14ee307dca8e954a8219c34484d7c4
-  set(CIVETWEB_CHECK_MD5_HASH 1b14ee307dca8e954a8219c34484d7c4 CACHE STRING
-    "The hash of Check unit testing framework archive to be downloaded")
-  set_property(CACHE CIVETWEB_CHECK_MD5_HASH PROPERTY VALUE ${CIVETWEB_CHECK_MD5_HASH})
-  mark_as_advanced(CIVETWEB_CHECK_MD5_HASH)
-
-  # Build the testing
-  add_subdirectory(unittest)
-endif()
-
 # Set up CPack
 include(InstallRequiredSystemLibraries)
 set(CPACK_PACKAGE_VENDOR "civetweb Contributors")
diff --git a/src/civetweb.c b/src/civetweb.c
index ee3d3e9..cc40747 100644
--- a/src/civetweb.c
+++ b/src/civetweb.c
@@ -10540,8 +10540,8 @@ forward_body_data(struct mg_connection *conn, FILE *fp, SOCKET sock, SSL *ssl)
 				    != nread) {
 					break;
 				}
+				conn->consumed_content += nread;
 			}
-			conn->consumed_content += nread;
 		}
 
 		if (conn->consumed_content == conn->content_len) {
@@ -15576,7 +15576,7 @@ ssl_get_protocol(int version_id)
  * definitions of this function, having a warning in one version or
  * another is unavoidable. */
 static void
-ssl_info_callback(SSL *ssl, int what, int ret)
+ssl_info_callback(const SSL *ssl, int what, int ret)
 {
 	(void)ret;
 
